### PR TITLE
A few warning fixes

### PR DESCRIPTION
**Pull request by bstoeger**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

These are a few warning fixes. They were either real bugs or at least questionable code.

Warning: untested! Another set of eyes would be great. For the fourth commit, we had a mailing list discussion without clear consensus.

I did not attempt to fix silly warnings (notably truncation of format strings) or code that was unclear to me.

I likewise didn't attempt to check the return of posix stylre read() - wouldn't we have to handle EINTR? Or would that be so absurdly unrealistic that we shouldn't bother? After all, it wasn't considered up to now.

### CODE REVIEW

**Comment by atdotde** [comment]
The first (gpstracker) I have no clue about the code to judge it. The rest looks good to me.

File: core/gpslocation.cpp [hunk 1]
```diff
@@ -436,13 +436,12 @@ void GpsLocation::deleteFixFromStorage(gpsTracker &gt)
 
 void GpsLocation::deleteGpsFix(qint64 when)
 {
-	struct gpsTracker defaultTracker;
-	defaultTracker.when = 0;
-	struct gpsTracker deletedTracker = m_trackers.value(when, defaultTracker);
-	if (deletedTracker.when != when) {
+	auto it = m_trackers.find(when);
+	if (it == m_trackers.end()) {
 		qDebug() << "can't find tracker for timestamp" << when;
```
Comment (neolit123) [inline core/gpslocation.cpp]: probably best to include `GpsLocation::deleteGpsFix():` to indicate the origin of the message.
`qWarning()` is probably a better choice too.

File: core/ostctools.c [hunk 2]
```diff
@@ -57,25 +57,37 @@ void ostctools_import(const char *file, struct dive_table *divetable)
 	// Read dive number from the log
 	uc_tmp = calloc(2, 1);
 	fseek(archive, 258, 0);
-	fread(uc_tmp, 1, 2, archive);
+	if (fread(uc_tmp, 1, 2, archive) != 2) {
+		report_error(translate("gettextFromC", "Failed to read '%s'"), file);
```
Comment (neolit123) [inline core/ostctools.c]: let's include this as a variable near the function  prologue to help the translation tooling:
`const char *failed_to_read = translate("gettextFromC", "Failed to read '%s'");`
and then re-use `failed_to_read`.

**Review by neolit123 — CHANGES_REQUESTED** [review]
thanks, lgtm aside from a couple of small comments.

File: core/ostctools.c [hunk 3]
```diff
@@ -57,25 +57,37 @@ void ostctools_import(const char *file, struct dive_table *divetable)
...
+	}
 	serial = uc_tmp[0] + (uc_tmp[1] << 8);
 	free(uc_tmp);
 
 	// Read dive's raw data, header + profile
 	fseek(archive, 456, 0);
-	while (!feof(archive)) {
-		fread(buffer + i, 1, 1, archive);
+	while ((c = getc(archive)) != EOF) {
+		buffer[i] = c;
```
Comment (neolit123) [inline core/ostctools.c:85]: i would probably add a sentence about the change to `getc()` from `fread()` in the commit message.

**Comment by dirkhh** [comment]
Thanks for doing these changes. I agree with @neolit123 's change requests, other than that it looks good to me.

File: core/gpslocation.cpp (hunk 1 above)
Comment (bstoeger) [reply core/gpslocation.cpp]: Is it OK to do this in the same commit?

Comment (neolit123) [reply core/gpslocation.cpp]: at think yes, as long as it mentions it in the commit message.

File: core/ostctools.c (hunk 2 above)
Comment (neolit123) [reply core/ostctools.c]: we should use `failed_to_read_msg` here.

File: core/ostctools.c [hunk 4]
```diff
@@ -57,25 +58,37 @@ void ostctools_import(const char *file, struct dive_table *divetable)
...
+	}
 	ostcdive->number = uc_tmp[0] + (uc_tmp[1] << 8);
 	free(uc_tmp);
 
 	// Read device's serial number
 	uc_tmp = calloc(2, 1);
 	fseek(archive, 265, 0);
-	fread(uc_tmp, 1, 2, archive);
+	if (fread(uc_tmp, 1, 2, archive) != 2) {
+		report_error(translate("gettextFromC", "Failed to read '%s'"), file);
```
Comment (neolit123) [inline core/ostctools.c]: we should use` failed_to_read_msg` here.

File: core/ostctools.c [hunk 5]
```diff
@@ -90,7 +103,7 @@ void ostctools_import(const char *file, struct dive_table *divetable)
 			dc_fam = DC_FAMILY_HW_OSTC3;
 			break;
 		default:
-			report_error(translate("gettextFromC", "Unknown DC in dive %d"), ostcdive->number);
+			report_error(failed_to_read_msg, ostcdive->number);
```
Comment (neolit123) [inline core/ostctools.c]: i think you switched the errors. these should not use `failed_to_read_msg`.

File: core/ostctools.c [hunk 6]
```diff
@@ -122,7 +135,7 @@ void ostctools_import(const char *file, struct dive_table *divetable)
 	// Prepare data to pass to libdivecomputer.
 	ret = ostc_prepare_data(model, dc_fam, devdata);
 	if (ret == 0) {
-		report_error(translate("gettextFromC", "Unknown DC in dive %d"), ostcdive->number);
+		report_error(failed_to_read_msg, ostcdive->number);
```
Comment (neolit123) [inline core/ostctools.c]: ditto.

**Review by neolit123 — CHANGES_REQUESTED** [review]

File: core/ostctools.c (hunk 6 above)
Comment (neolit123) [reply core/ostctools.c]: maybe a separate var for these - e.g. `uknown_dc_msg`?

**Comment by bstoeger** [comment]
Gaah.. Sorry, too tired already. Should work now.

**Comment by neolit123** [comment]
thanks, lgtm. waiting on CI.

**Review by neolit123 — APPROVED** [review]

**Pull request merged**